### DISCUSSION TITLE
TTS questions

### DISCUSSION THREAD

**Discussion by Sptzzz** [n3d1117/chatgpt-telegram-bot#570] [Q&A]
Hi all,

I managed to get gpt-4o working here (pretty simple really) but the main thing I'm trying to get working differently is /tts 

It seems that right now if you prompt for example: `/tts a cat meowing` it will just output literally "a cat meowing".

using the gtts plugin you can just say `tts a snare drum solo` and it will, hilariously, output something like "badum ts ts" etc.

Is there a way for the native /tts command to function in the same manner?

**Comment by gilcu3** [comment]
Hi, I think the original aim of the `/tts` command was to literally convert text to speech, similar to the `/image` command that converts a prompt to an image. To make `chatgpt` reason about the `tts` to be generated, you would rather use the plugins as you mention. In which use case would this not fit?

**Reply by Sptzzz** [reply]
Because the plugins are google tts, not the vastly superior openai's tts. :/

**Reply by gilcu3** [reply]
> Because the plugins are google tts, not the vastly superior openai's tts. :/

Then that should be doable already, check out the `auto-tts` plugin added in https://github.com/n3d1117/chatgpt-telegram-bot/pull/468

**Reply by Sptzzz** [reply]
Yeah I just did. It's still very spotty. Like, if you say "send a recording of a dog barking" the voice message will be: "here's a dog barking." and that's it loool

**Comment by OKyryllov** [comment]
How did you managed to make gpt-4o work?

**Reply by Sptzzz** [reply]
https://github.com/n3d1117/chatgpt-telegram-bot/pull/569
